### ISSUE TITLE
failed to set piecewise_decay param in paralleldo mode

### ISSUE THREAD

**Issue by walloollaw** [PaddlePaddle/Paddle#11196]
i failed to set `piecewise_decay` in parallel training mode with special params using the code in `https://github.com/PaddlePaddle/models/blob/develop/fluid/image_classification/train.py`

normal case: (both ok run on single gpu or multiple gpus)
```
    bd = [500, 1000, 4000]
    lr = [0.01, 0.01, 0.001, 0.0001]
    optimizer = fluid.optimizer.Momentum(
                learning_rate=fluid.layers.piecewise_decay(
                boundaries=bd, values=lr),
            momentum=0.9,
            regularization=fluid.regularizer.L2Decay(1e-4))
````

abnormal case:(failed on multiple gpus but ok on single gpu)
```
    bd = [1000, 4000]
    lr = [0.01, 0.001, 0.0001]
    optimizer = fluid.optimizer.Momentum(
                learning_rate=fluid.layers.piecewise_decay(
                boundaries=bd, values=lr),
            momentum=0.9,
            regularization=fluid.regularizer.L2Decay(1e-4))
````

the errors:
```
The default exception handler is LOG(FATAL).holder_ should not be null
Tensor not initialized yet when Tensor::type() is called. at [/paddle/new_paddle/Paddle/paddle/fluid/framework/tensor.h:119]
PaddlePaddle Call Stacks: 
0       0x7fb27c67aa56p paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int) + 486
1       0x7fb27c67d1d6p paddle::framework::Tensor::type() const + 150
2       0x7fb27d1cdc55p paddle::framework::OperatorWithKernel::IndicateDataType(paddle::framework::ExecutionContext const&) const + 117
3       0x7fb27d1cdf8fp paddle::framework::OperatorWithKernel::GetExpectedKernelType(paddle::framework::ExecutionContext const&) const + 47
4       0x7fb27d1ce67bp paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const + 235
5       0x7fb27c7095b0p paddle::framework::Executor::RunPreparedContext(paddle::framework::ExecutorPrepareContext*, paddle::framework::Scope*, bool, bool) + 336
6       0x7fb27c709d14p paddle::framework::Executor::Run(paddle::framework::ProgramDesc const&, paddle::framework::Scope*, int, bool, bool) + 100
7       0x7fb27cfde8cdp std::_Function_handler<std::unique_ptr<paddle::platform::EnforceNotMet, std::default_delete<paddle::platform::EnforceNotMet> > (), std::reference_wrapper<std::_Bind_simple<std::reference_wrapper<std::future<std::unique_ptr<paddle::platform::EnforceNotMet, std::default_delete<paddle::platform::EnforceNotMet> > > paddle::framework::ThreadPool::RunAndGetException<paddle::operators::ParallelDoGradOp::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const::{lambda()#1}>(paddle::operators::ParallelDoGradOp::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const::{lambda()#1})::{lambda()#1}> ()> > >::_M_invoke(std::_Any_data const&) + 93
8       0x7fb27cfdc0bap std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<std::unique_ptr<paddle::platform::EnforceNotMet, std::default_delete<paddle::platform::EnforceNotMet> > >, std::__future_base::_Result_base::_Deleter>, std::unique_ptr<paddle::platform::EnforceNotMet, std::default_delete<paddle::platform::EnforceNotMet> > > >::_M_invoke(std::_Any_data const&) + 42
9       0x7fb27cfdb6d7p std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&) + 39
10      0x7fb2b87d6973p pthread_once + 83
11      0x7fb27cfe4364p std::__future_base::_Task_state<std::future<std::unique_ptr<paddle::platform::EnforceNotMet, std::default_delete<std::unique_ptr> > > paddle::framework::ThreadPool::RunAndGetException<paddle::operators::ParallelDoGradOp::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const::{lambda()#1}>(paddle::operators::ParallelDoGradOp::RunImpl(paddle::framework::Scope const&, boost::variant<paddle::platform::CUDAPlace, paddle::platform::CPUPlace, paddle::platform::CUDAPinnedPlace, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> const&) const::{lambda()#1})::{lambda()#1}, std::allocator<int>, std::default_delete<std::unique_ptr> ()>::_M_run() + 324
12      0x7fb27d1e3d18p paddle::framework::ThreadPool::TaskLoop() + 920
13      0x7fb2a85aa8a0p
14      0x7fb2b87d11c3p
15      0x7fb2b776212dp clone + 109
*** Check failure stack trace: ***
    @     0x7fb27e94c14d  google::LogMessage::Fail()
    @     0x7fb27e94fbfc  google::LogMessage::SendToLog()
    @     0x7fb27e94bc73  google::LogMessage::Flush()
    @     0x7fb27e95110e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fb27cfde3c8  std::_Function_handler<>::_M_invoke()
    @     0x7fb27cfdaf43  std::_Function_handler<>::_M_invoke()
    @     0x7fb27cfdb6d7  std::__future_base::_State_base::_M_do_set()
    @     0x7fb2b87d6973  __GI___pthread_once
    @     0x7fb27cfdbda9  std::__future_base::_Deferred_state<>::_M_run_deferred()
    @     0x7fb27cfe3cd9  paddle::operators::ParallelDoGradOp::RunImpl()
    @     0x7fb27c7095b0  paddle::framework::Executor::RunPreparedContext()
    @     0x7fb27c709d14  paddle::framework::Executor::Run()
    @     0x7fb27c6920fb  _ZZN8pybind1112cpp_function10initializeIZNS0_C1IvN6paddle9framework8ExecutorEJRKNS4_11ProgramDescEPNS4_5ScopeEibbEJNS_4nameENS_9is_methodENS_7siblingEEEEMT0_FT_DpT1_EDpRKT2_EUlPS5_S8_SA_ibbE_vJSO_S8_SA_ibbEJSB_SC_SD_EEEvOSF_PFSE_SH_ESN_ENUlRNS_6detail13function_callEE1_4_FUNESV_
    @     0x7fb27c68bb54  pybind11::cpp_function::dispatcher()
    @     0x7fb2b8af4f46  PyEval_EvalFrameEx
    @     0x7fb2b8af597d  PyEval_EvalCodeEx
    @     0x7fb2b8af4e65  PyEval_EvalFrameEx
    @     0x7fb2b8af597d  PyEval_EvalCodeEx
    @     0x7fb2b8af4e65  PyEval_EvalFrameEx
    @     0x7fb2b8af597d  PyEval_EvalCodeEx
    @     0x7fb2b8af5ab2  PyEval_EvalCode
    @     0x7fb2b8b20612  PyRun_FileExFlags
    @     0x7fb2b8b21b49  PyRun_SimpleFileExFlags
    @     0x7fb2b8b3757d  Py_Main
    @     0x7fb2b769cbd5  __libc_start_main
    @           0x4007e1  (unknown)
    @              (nil)  (unknown)
Aborted
```

**Comment by jacquesqiao** [comment]
why normal case and abnormal case both have `multiple gpus`, and does normal case mean it can run?

**Comment by jacquesqiao** [comment]
set
```
export GLOG_v=3
export GLOG_logtostderr=1
```
to see more logs.

**Comment by walloollaw** [comment]
after set
```
export GLOG_v=3
export GLOG_logtostderr=1
```
not found any useful info, detail log has been uploaded [here](https://pan.baidu.com/s/1MdGO8svQ9hibvTczMkrv1Q)

**Comment by kebinC** [comment]
I also have this problem when setting 
    epoch_points = [5, 10]
    lr = [0.1, 0.01, 0.001]
, and it solved by setting
    epoch_points = [5, 10, 15]
    lr = [0.1, 0.01, 0.001, 0.0001]

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
